Sources/System/FilePath/FilePathSyntax.swift

```diff
@@ -0,0 +1,353 @@
+// Query API
+// @available(...)
+extension FilePath {
+  /// Returns true if this path uniquely identifies the location of
+  /// a file without reference to an additional starting location.
+  ///
+  /// On Unix platforms, absolute paths begin with a `/`. `isAbsolute` is equivalent
+  /// to `root != nil`.
+  ///
+  /// On Windows, absolute paths are fully qualified paths. UNC paths and device paths
+  /// are always absolute. Traditional DOS paths are absolute if they begin with a volume or drive
+  /// followed by a `:` and a separator.
+  ///
+  /// NOTE: This does not perform shell expansion or substitute
+  /// environment variables; paths beginning with `~` are considered relative.
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `/usr/local/bin`
+  ///   * `/tmp/foo.txt`
+  ///   * `/`
+  /// * Windows:
+  ///   * `C:\Users\`
+  ///   * `\\?\UNC\server\share\bar.exe`
+  ///   * `\\server\share\bar.exe`
+  ///
+  /// TODO: We should add API to system to access environment variables
+  /// and perform expansion/substituion, but that's deferred for now. Should
+  /// we mention that?
+  public var isAbsolute: Bool {
+    self.root?._rootIsAbsolute ?? false
+  }
+
+  /// Returns true if this path is not absolute (see absolute)
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `~/bar`
+  ///   * `tmp/foo.txt`
+  /// * Windows:
+  ///   * `bar\baz`
+  ///   * `C:Users\`
+  ///   * `\Users`
+  public var isRelative: Bool { !isAbsolute }
```

Comment:
I'm not sure this is needed, since it's unambiguously defined as "not absolute".

Reply:
I get what you're saying. I think it's a useful aid to reduce cognitive burden because "relative" is a concept that is used elsewhere, even though it is trivially derivable.

Reply:
That makes sense.